python/dgl/dataloading/dataloader.py

```diff
@@ -825,6 +827,26 @@ def __init__(self, graph, indices, graph_sampler, device=None, use_ddp=False,
 
         self.other_storages = {}
 
+        if use_cpu_worker_affinity:
+            nw_work = kwargs.get('num_workers', 0)
+
+            if cpu_worker_affinity_cores is None:
+                cpu_worker_affinity_cores = []
+
+            if not isinstance(cpu_worker_affinity_cores, list):
+                raise Exception('ERROR: cpu_worker_affinity_cores should be a list of cores')
+            if not nw_work > 0:
+                raise Exception('ERROR: affinity should be used with --num_workers=X')
+            if len(cpu_worker_affinity_cores) not in [0, nw_work]:
+                raise Exception('ERROR: cpu_affinity incorrect '
+                                'settings for cores={} num_workers={}'
+                                .format(cpu_worker_affinity_cores, nw_work))
+
+            self.cpu_cores = (cpu_worker_affinity_cores
+                                if len(cpu_worker_affinity_cores)
+                                else range(0, nw_work))
+            worker_init_fn = self.worker_init_function
```

Comment:
This will overwrite `worker_init_fn` above, so it disables the `WorkerInitWrapper` above that sets the number of OpenMP threads per process, and it will not invoke the custom `worker_init_fn`.  Is this intended?  If not, you could put the logic in `WorkerInitWrapper`.